packages/salesforcedx-vscode-core/test/jest/conflict/timestampConflictDetector.test.ts

```diff
@@ -0,0 +1,745 @@
+      cachePath:
+        '/var/folders/71/2811gwnx2n7fwqyyf99ctsfm0000gn/T/.sfdx/diff/test-ph2qxpndpvi7@example.com/metadataPackage_1679151464361/main/default/classes/TestApex22.cls'
+    },
+    {
+      projectPath:
+        '/Users/kenneth.lewis/scratchpad/TestProject-1/force-app/main/default/classes/TestApex22.cls-meta.xml',
+      cachePath:
+        '/var/folders/71/2811gwnx2n7fwqyyf99ctsfm0000gn/T/.sfdx/diff/test-ph2qxpndpvi7@example.com/metadataPackage_1679151464361/main/default/classes/TestApex22.cls-meta.xml'
+    }
+  ];
+  let persistentStorageServiceMock: jest.SpyInstance;
+  let correlateResultsStub: jest.SpyInstance;
+  let diffComponentsStub: jest.SpyInstance;
+
+  describe('createDiffs', () => {
+    beforeEach(() => {
+      correlateResultsStub = jest
+        .spyOn(MetadataCacheService, 'correlateResults')
+        .mockReturnValue(dummyCorrelatedComponents as any);
+      persistentStorageServiceMock = jest
+        .spyOn(PersistentStorageService, 'getInstance')
+        .mockReturnValue({
+          makeKey: jest.fn(),
+          getPropertiesForFile: jest.fn().mockResolvedValueOnce({
+            // For the first file, give a value that will trip the
+            // conflict checker
+            lastModifiedDate: dummyLastModifiedDateLocal
+          })
+        } as any);
+    });
+
+    it('should return diff results for only the files that trip the timestamp conflict detector', async () => {
+      // Only return a diff for the first file, which should have tripped the timestamp check
+      diffComponentsStub = jest
+        .spyOn(diffUtils, 'diffComponents')
+        .mockReturnValueOnce(dummyDiffs)
+        .mockReturnValueOnce([]);
+      const timestampConflictDetector = new TimestampConflictDetector();
+
+      const diffs = timestampConflictDetector.createDiffs(
+        dummyMetadataCacheResult as any
+      );
+
+      expect(correlateResultsStub).toHaveBeenCalledWith(
+        dummyMetadataCacheResult
+      );
+      expect(persistentStorageServiceMock).toHaveBeenCalled();
+      expect(diffComponentsStub).toHaveBeenCalledTimes(2);
+      expect(diffs.different.size).toBe(2);
+    });
+
+    it('should return diff results for all files passed in when the skipTimestampCheck option is used', async () => {
+      diffComponentsStub = jest
+        .spyOn(diffUtils, 'diffComponents')
+        .mockReturnValueOnce(dummyDiffs)
+        .mockReturnValueOnce(dummyDiffs2);
+      const timestampConflictDetector = new TimestampConflictDetector();
+
+      const diffs = timestampConflictDetector.createDiffs(
+        dummyMetadataCacheResult as any,
+        true
+      );
+
+      expect(correlateResultsStub).toHaveBeenCalledWith(
+        dummyMetadataCacheResult
+      );
+      expect(persistentStorageServiceMock).toHaveBeenCalled();
+      expect(diffComponentsStub).toHaveBeenCalledTimes(2);
+      expect(diffs.different.size).toBe(4);
+    });
+  });
```

Comment:
What would you think about adding 1 more test that returns no results when the timestamp conflict detector is not tripped?

Reply:
Currently, I think this is covered in one of the integration tests. I would love to fully port, does integration test over into this unit test file, but would also love to do it as a follow up work item. :-)

<img width="450" alt="2023-03-23_16-53-39" src="https://user-images.githubusercontent.com/46458081/227389614-041e577b-61c2-4ff5-b44b-c37096978fad.png">